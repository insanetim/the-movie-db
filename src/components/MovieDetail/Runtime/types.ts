import { IMovieDetail } from 'src/interfaces/movie.interface'

export type RuntimeProps = {
  runtime: IMovieDetail['runtime']
}
