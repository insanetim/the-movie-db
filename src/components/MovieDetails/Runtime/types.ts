import { IMovieDetails } from 'src/interfaces/movie.interface'

export type RuntimeProps = {
  runtime: IMovieDetails['runtime']
}
