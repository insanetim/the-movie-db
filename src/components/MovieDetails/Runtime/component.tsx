import { Col, Typography } from 'antd'
import convertDuration from 'src/utils/convertDataHelpers/convertDuration'

import { RuntimeProps } from './types'

const Runtime: React.FC<RuntimeProps> = ({ runtime }) => (
  <Col span={24}>
    <Typography.Paragraph>
      <b>Runtime: </b>
      <span>{convertDuration(runtime)}</span>
    </Typography.Paragraph>
  </Col>
)

export default Runtime
