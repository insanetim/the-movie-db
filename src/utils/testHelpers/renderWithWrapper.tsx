import { render, renderHook, RenderOptions } from '@testing-library/react'
import { PropsWithChildren, ReactNode, Suspense } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { AppStore, RootState, setupStore } from 'src/store'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: RootState
  store?: AppStore
}

const createWrapper =
  (store: AppStore) =>
  ({ children }: PropsWithChildren) => (
    <Provider store={store}>
      <Suspense fallback={null}>
        <BrowserRouter>{children}</BrowserRouter>
      </Suspense>
    </Provider>
  )

export const renderWithWrapper = (
  ui: ReactNode,
  {
    preloadedState = {} as RootState,
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = createWrapper(store)
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export const renderHookWithWrapper = <T,>(
  hook: () => T,
  {
    preloadedState = {} as RootState,
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = createWrapper(store)
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
