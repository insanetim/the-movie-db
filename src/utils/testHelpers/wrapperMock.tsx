import { ReactNode } from 'react'
import { BrowserRouter } from 'react-router-dom'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <BrowserRouter>{children}</BrowserRouter>
}
export default Wrapper
