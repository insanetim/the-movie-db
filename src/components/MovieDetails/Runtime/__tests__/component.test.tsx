import { screen } from '@testing-library/react'
import { renderWithWrapper } from 'src/utils/testHelpers/renderWithWrapper'

import Runtime from '../component'

describe('Runtime component', () => {
  it('renders runtime duration in hours and minutes', () => {
    renderWithWrapper(<Runtime runtime={150} />)

    expect(screen.getByText('Runtime:')).toBeInTheDocument()
    expect(screen.getByText('2h 30m')).toBeInTheDocument()
  })
})
