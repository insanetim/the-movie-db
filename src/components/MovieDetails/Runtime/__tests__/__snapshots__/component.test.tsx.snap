// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Runtime component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-24 css-dev-only-do-not-override-11xg00t"
  >
    <div
      class="ant-typography css-dev-only-do-not-override-11xg00t"
    >
      <b>
        Runtime: 
      </b>
      <span>
        2h 30m
      </span>
    </div>
  </div>
</DocumentFragment>
`;
